x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java

```diff
@@ -66,6 +66,8 @@
         CloseIndexAction.NAME + "*");
     private static final Automaton MANAGE_LEADER_INDEX_AUTOMATON = patterns(ForgetFollowerAction.NAME + "*");
     private static final Automaton MANAGE_ILM_AUTOMATON = patterns("indices:admin/ilm/*");
+    private static final Automaton MAINTENANCE_AUTOMATON = patterns("indices:admin/refresh", "indices:admin/flush",
```

Comment:
The test that it's failing `{"type":"security_exception","reason":"action [indices:admin/refresh[s]] is unauthorized for user [u15]"}}]}}` hints that `indices:admin/refresh[s]` is unauthorized.
The refresh action translates into shard wise sub-actions, which are also authorized by themselves.
You need to add replace `indices:admin/refresh` with `indices:admin/refresh*`.

Reply:
This should be:
```
private static final Automaton MAINTENANCE_AUTOMATON = patterns("indices:admin/refresh*", "indices:admin/flush*", "indices:admin/synced_flush", "indices:admin/forcemerge*");
```
The motive is that, as mentioned above, a cluster action might spawn other actions, both on the node first handling the request as well as other nodes on the cluster. Every such action is authorized individually, i.e. the authorization status of the parent action is not consulted. For example, for the `refresh` (and `flush`) action translate to `indices:admin/refresh[s]` (and `indices:admin/flush[s]`) actions for every index shard, which are then translated into `indices:admin/refresh[p]` and `indices:admin/refresh[r]` for the primary and replica shards. The `maintenance` privilege should grant all these action types, and we customary add the wildcard star to the name of the originating action instead of iterating all the sub-action names.